sklearn/utils/testing.py

```diff
@@ -372,14 +375,30 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _assert_sparse_allclose(actual, desired, rtol=1e-7, atol=0, err_msg='',
```

Comment:
Yes, either of those solutions are acceptable to me, while a complicated test with many paths inline is not acceptable.